Failure building the iOS demo app from the command line

The build fails when trying to build the iOS demo app from the command line.

Steps to reproduce:
1. Do a typical `npm ci` in the gb-mobile root to install the NPM packages from scratch
2. In a second terminal window, run `npm run start:reset` to launch Metro
3. In the first terminal window, do `npm run core ios` to build and launch the iOS demo app in a simulator
4. The build will generally go a long way, the simulator will open but, in the stage where the app would be installed the following error occurs:

```
info Found Xcode workspace "gutenberg.xcworkspace"
info Building (using "xcodebuild -workspace gutenberg.xcworkspace -configuration Debug -scheme gutenberg -destination id=860DC2DE-1126-4395-98E2-B5A89E804633 -derivedDataPath build/gutenberg")
...............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
info Installing "DerivedData/Build/Products/Debug-iphonesimulator/gutenberg.app"
An error was encountered processing the command (domain=NSPOSIXErrorDomain, code=2):
Failed to install the requested application
An application bundle was not found at the provided path.
Provide a valid path to the desired application bundle.
Print: Entry, ":CFBundleIdentifier", Does Not Exist
error Command failed: /usr/libexec/PlistBuddy -c Print:CFBundleIdentifier DerivedData/Build/Products/Debug-iphonesimulator/gutenberg.app/Info.plist
Print: Entry, ":CFBundleIdentifier", Does Not Exist
. Run CLI with --verbose flag for more details.
Error: Command failed: /usr/libexec/PlistBuddy -c Print:CFBundleIdentifier DerivedData/Build/Products/Debug-iphonesimulator/gutenberg.app/Info.plist
Print: Entry, ":CFBundleIdentifier", Does Not Exist

    at checkExecSyncError (child_process.js:630:11)
    at Object.execFileSync (child_process.js:648:15)
    at runOnSimulator (/Users/stefanos/proj/a8c/rn/t1/gutenberg-mobile/gutenberg/node_modules/@react-native-community/cli-platform-ios/build/commands/runIOS/index.js:183:45)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Command.handleAction (/Users/stefanos/proj/a8c/rn/t1/gutenberg-mobile/gutenberg/node_modules/react-native/node_modules/@react-native-community/cli/build/index.js:182:9)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @wordpress/react-native-editor@1.37.0 ios: `react-native run-ios`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @wordpress/react-native-editor@1.37.0 ios script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/stefanos/.npm/_logs/2020-09-21T10_53_26_741Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gutenberg@9.0.0 native: `npm run --prefix packages/react-native-editor "ios"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the gutenberg@9.0.0 native script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/stefanos/.npm/_logs/2020-09-21T10_53_26_781Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gutenberg-mobile@1.36.0 core: `cd gutenberg && npm run native "ios"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the gutenberg-mobile@1.36.0 core script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/stefanos/.npm/_logs/2020-09-21T10_53_26_808Z-debug.log
```
5. Error doesn't happen if you open the demo project on Xcode directly and build+run from there

On macos 10.15.6, Xcode v12.0

--- a/.github/workflows/rnmobile-ios-runner.yml
+++ b/.github/workflows/rnmobile-ios-runner.yml
@@ -30,7 +30,7 @@ jobs:
     - name: Restore build cache
       uses: actions/cache@v2
       with:
-        path: packages/react-native-editor/ios/build/gutenberg/Build/Products/Release-iphonesimulator/GutenbergDemo.app
+        path: packages/react-native-editor/ios/build/gutenberg/Build/Products/Release-iphonesimulator/gutenberg.app
         key: ${{ runner.os }}-ios-build-${{ hashFiles('ios-checksums.txt') }}
 
     - name: Restore pods cache
@@ -55,13 +55,13 @@ jobs:
       run: sudo xcode-select --switch /Applications/Xcode_11.4.1.app
 
     - name: Build (if needed)
-      run: test -e packages/react-native-editor/ios/build/gutenberg/Build/Products/Release-iphonesimulator/GutenbergDemo.app/GutenbergDemo || SKIP_BUNDLING=true npm run native test:e2e:build-app:ios
+      run: test -e packages/react-native-editor/ios/build/gutenberg/Build/Products/Release-iphonesimulator/gutenberg.app/gutenberg || SKIP_BUNDLING=true npm run native test:e2e:build-app:ios
 
     - name: Run iOS Device Tests
       run: TEST_RN_PLATFORM=ios npm run native device-tests:local  ${{ matrix.native-test-name }}
 
     - name: Prepare build cache
-      run: rm packages/react-native-editor/ios/build/gutenberg/Build/Products/Release-iphonesimulator/GutenbergDemo.app/main.jsbundle
+      run: rm packages/react-native-editor/ios/build/gutenberg/Build/Products/Release-iphonesimulator/gutenberg.app/main.jsbundle
 
     - uses: actions/upload-artifact@v2
       if: always()

--- a/packages/react-native-editor/__device-tests__/helpers/utils.js
+++ b/packages/react-native-editor/__device-tests__/helpers/utils.js
@@ -29,7 +29,7 @@ const testEnvironment = process.env.TEST_ENV || defaultEnvironment;
 const defaultAndroidAppPath =
 	'./android/app/build/outputs/apk/debug/app-debug.apk';
 const defaultIOSAppPath =
-	'./ios/build/gutenberg/Build/Products/Release-iphonesimulator/GutenbergDemo.app';
+	'./ios/build/gutenberg/Build/Products/Release-iphonesimulator/gutenberg.app';
 
 const localAndroidAppPath =
 	process.env.ANDROID_APP_PATH || defaultAndroidAppPath;

--- a/packages/react-native-editor/ios/gutenberg.xcodeproj/project.pbxproj
+++ b/packages/react-native-editor/ios/gutenberg.xcodeproj/project.pbxproj
@@ -57,7 +57,7 @@
 		034874811302E030BA1637A1 /* Pods-gutenbergTests.debug.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-gutenbergTests.debug.xcconfig"; path = "Target Support Files/Pods-gutenbergTests/Pods-gutenbergTests.debug.xcconfig"; sourceTree = "<group>"; };
 		083D3CAD9B26701AE0659EEB /* Pods-gutenberg.release.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-gutenberg.release.xcconfig"; path = "Target Support Files/Pods-gutenberg/Pods-gutenberg.release.xcconfig"; sourceTree = "<group>"; };
 		0EB766FE2F6D446A80AC6E6A /* libRNSVG.a */ = {isa = PBXFileReference; explicitFileType = undefined; fileEncoding = 9; includeInIndex = 0; lastKnownFileType = archive.ar; path = libRNSVG.a; sourceTree = "<group>"; };
-		13B07F961A680F5B00A75B9A /* GutenbergDemo.app */ = {isa = PBXFileReference; explicitFileType = wrapper.application; includeInIndex = 0; path = GutenbergDemo.app; sourceTree = BUILT_PRODUCTS_DIR; };
+		13B07F961A680F5B00A75B9A /* gutenberg.app */ = {isa = PBXFileReference; explicitFileType = wrapper.application; includeInIndex = 0; path = gutenberg.app; sourceTree = BUILT_PRODUCTS_DIR; };
 		13B07FB21A68108700A75B9A /* Base */ = {isa = PBXFileReference; lastKnownFileType = file.xib; name = Base; path = Base.lproj/LaunchScreen.xib; sourceTree = "<group>"; };
 		13B07FB51A68108700A75B9A /* Images.xcassets */ = {isa = PBXFileReference; lastKnownFileType = folder.assetcatalog; name = Images.xcassets; path = gutenberg/Images.xcassets; sourceTree = "<group>"; };
 		13B07FB61A68108700A75B9A /* Info.plist */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.xml; name = Info.plist; path = gutenberg/Info.plist; sourceTree = "<group>"; };
@@ -218,7 +218,7 @@
 		83CBBA001A601CBA00E9B192 /* Products */ = {
 			isa = PBXGroup;
 			children = (
-				13B07F961A680F5B00A75B9A /* GutenbergDemo.app */,
+				13B07F961A680F5B00A75B9A /* gutenberg.app */,
 				00E356EE1AD99517003FC87E /* gutenbergTests.xctest */,
 				2D02E47B1E0B4A5D006451C7 /* gutenberg-tvOS.app */,
 				2D02E4901E0B4A5D006451C7 /* gutenberg-tvOSTests.xctest */,
@@ -312,7 +312,7 @@
 			);
 			name = gutenberg;
 			productName = "Hello World";
-			productReference = 13B07F961A680F5B00A75B9A /* GutenbergDemo.app */;
+			productReference = 13B07F961A680F5B00A75B9A /* gutenberg.app */;
 			productType = "com.apple.product-type.application";
 		};
 		2D02E47A1E0B4A5D006451C7 /* gutenberg-tvOS */ = {
@@ -823,7 +823,7 @@
 					"-lc++",
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = org.wordpress.gutenberg.development;
-				PRODUCT_NAME = GutenbergDemo;
+				PRODUCT_NAME = gutenberg;
 				PROVISIONING_PROFILE_SPECIFIER = "";
 				SWIFT_OBJC_BRIDGING_HEADER = "gutenberg-Bridging-Header.h";
 				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
@@ -857,7 +857,7 @@
 					"-lc++",
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = org.wordpress.gutenberg.development;
-				PRODUCT_NAME = GutenbergDemo;
+				PRODUCT_NAME = gutenberg;
 				PROVISIONING_PROFILE_SPECIFIER = "";
 				SWIFT_OBJC_BRIDGING_HEADER = "gutenberg-Bridging-Header.h";
 				SWIFT_VERSION = 5.0;

--- a/packages/react-native-editor/ios/gutenberg.xcodeproj/xcshareddata/xcschemes/gutenberg.xcscheme
+++ b/packages/react-native-editor/ios/gutenberg.xcodeproj/xcshareddata/xcschemes/gutenberg.xcscheme
@@ -29,7 +29,7 @@
             <BuildableReference
                BuildableIdentifier = "primary"
                BlueprintIdentifier = "13B07F861A680F5B00A75B9A"
-               BuildableName = "GutenbergDemo.app"
+               BuildableName = "gutenberg.app"
                BlueprintName = "gutenberg"
                ReferencedContainer = "container:gutenberg.xcodeproj">
             </BuildableReference>
@@ -59,7 +59,7 @@
          <BuildableReference
             BuildableIdentifier = "primary"
             BlueprintIdentifier = "13B07F861A680F5B00A75B9A"
-            BuildableName = "GutenbergDemo.app"
+            BuildableName = "gutenberg.app"
             BlueprintName = "gutenberg"
             ReferencedContainer = "container:gutenberg.xcodeproj">
          </BuildableReference>
@@ -92,7 +92,7 @@
          <BuildableReference
             BuildableIdentifier = "primary"
             BlueprintIdentifier = "13B07F861A680F5B00A75B9A"
-            BuildableName = "GutenbergDemo.app"
+            BuildableName = "gutenberg.app"
             BlueprintName = "gutenberg"
             ReferencedContainer = "container:gutenberg.xcodeproj">
          </BuildableReference>
@@ -109,7 +109,7 @@
          <BuildableReference
             BuildableIdentifier = "primary"
             BlueprintIdentifier = "13B07F861A680F5B00A75B9A"
-            BuildableName = "GutenbergDemo.app"
+            BuildableName = "gutenberg.app"
             BlueprintName = "gutenberg"
             ReferencedContainer = "container:gutenberg.xcodeproj">
          </BuildableReference>

--- a/packages/react-native-editor/package.json
+++ b/packages/react-native-editor/package.json
@@ -114,7 +114,7 @@
     "test:e2e:bundle:android": "mkdir -p android/app/src/main/assets && npm run rn-bundle -- --reset-cache --platform android --dev false --minify false --entry-file index.js --bundle-output android/app/src/main/assets/index.android.bundle --assets-dest android/app/src/main/res",
     "test:e2e:build-app:android": "npm run test:e2e:bundle:android && cd android && ./gradlew clean && ./gradlew assembleDebug",
     "test:e2e:install-app:android": "cd android && ./gradlew installDebug",
-    "test:e2e:bundle:ios": "mkdir -p ios/build/gutenberg/Build/Products/Release-iphonesimulator/GutenbergDemo.app && npm run rn-bundle -- --reset-cache --platform=ios --dev=false --minify false --entry-file=index.js --bundle-output=./ios/build/gutenberg/Build/Products/Release-iphonesimulator/GutenbergDemo.app/main.jsbundle --assets-dest=./ios/build/gutenberg/Build/Products/Release-iphonesimulator/GutenbergDemo.app",
+    "test:e2e:bundle:ios": "mkdir -p ios/build/gutenberg/Build/Products/Release-iphonesimulator/gutenberg.app && npm run rn-bundle -- --reset-cache --platform=ios --dev=false --minify false --entry-file=index.js --bundle-output=./ios/build/gutenberg/Build/Products/Release-iphonesimulator/gutenberg.app/main.jsbundle --assets-dest=./ios/build/gutenberg/Build/Products/Release-iphonesimulator/gutenberg.app",
     "test:e2e:build-app:ios": "npm run ios -- --configuration Release --no-packager --simulator 'iPhone 11 (13.4)'",
     "build:gutenberg": "cd gutenberg && npm ci && npm run build",
     "clean": "npm run clean:build-artifacts; npm run clean:aztec; npm run clean:haste; npm run clean:jest; npm run clean:metro; npm run clean:react; npm run clean:watchman",

